File: android/src/main/kotlin/com/techlabs/flutter/flutter_techlabs_library/lib/model/AppVerInfo.kt
package com.techlabs.flutter.flutter_techlabs_library.lib.model

import org.json.JSONObject

/**
 * S3 서비스 정보 JSON의 appVerInfo 배열 항목 모델.
 */
data class AppVerInfo(
    val device: String,
    val appName: String,
    val ver: String,
    val type: String,
    val appName2: String,
    val latestVer: String
) {
    companion object {
        fun fromJsonObject(jsonObject: JSONObject): AppVerInfo {
            return AppVerInfo(
                device = jsonObject.optString("device", ""),
                appName = jsonObject.optString("app_name", ""),
                ver = jsonObject.optString("ver", ""),
                type = jsonObject.optString("type", ""),
                appName2 = jsonObject.optString("app_name2", ""),
                latestVer = jsonObject.optString("latest_ver", "")
            )
        }
    }

    fun toVersionCheckInfo(): VersionCheckInfo {
        return VersionCheckInfo(
            ver = ver,
            latestVer = latestVer,
            type = type,
            appName2 = appName2
        )
    }
}

File: lib/src/models/app_ver_info.dart
import 'version_check_info.dart';

class AppVerInfo {
  final String device;
  final String appName;
  final String ver;
  final String type;
  final String appName2;
  final String latestVer;

  const AppVerInfo({
    required this.device,
    required this.appName,
    required this.ver,
    required this.type,
    required this.appName2,
    required this.latestVer,
  });

  factory AppVerInfo.fromMap(Map<Object?, Object?> map) {
    return AppVerInfo(
      device: (map['device'] as String?) ?? '',
      appName: (map['app_name'] as String?) ?? '',
      ver: (map['ver'] as String?) ?? '',
      type: (map['type'] as String?) ?? '',
      appName2: (map['app_name2'] as String?) ?? '',
      latestVer: (map['latest_ver'] as String?) ?? '',
    );
  }

  Map<String, Object?> toMap() {
    return {
      'device': device,
      'app_name': appName,
      'ver': ver,
      'type': type,
      'app_name2': appName2,
      'latest_ver': latestVer,
    };
  }

  VersionCheckInfo toVersionCheckInfo() {
    return VersionCheckInfo(
      ver: ver,
      latestVer: latestVer,
      type: type,
      appName2: appName2,
    );
  }
}
